src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -163,34 +165,143 @@ private async Task CreateTable(string tableName, List<KeySchemaElement> keys, Li
+                    || request.ProvisionedThroughput?.WriteCapacityUnits != tableDescription.ProvisionedThroughput?.WriteCapacityUnits)
                 {
-                    description = await GetTableDescription(tableName);
+                    await ddbClient.UpdateTableAsync(request);
+                    tableDescription = await TableWaitOnStatusAsync(tableDescription.TableName, TableStatus.UPDATING, TableStatus.ACTIVE);
+                }
 
-                    await Task.Delay(2000);
+                await TableUpdateTtlAsync(tableDescription.TableName, ttlAttributeName);
 
-                } while (description.TableStatus == TableStatus.CREATING);
+                var existingGlobalSecondaryIndexes = tableDescription.GlobalSecondaryIndexes.Select(globalSecondaryIndex => globalSecondaryIndex.IndexName).ToArray();
+                var secondaryIndexesToCreate = (secondaryIndexes ?? Enumerable.Empty<GlobalSecondaryIndex>()).Where(secondaryIndex => !existingGlobalSecondaryIndexes.Contains(secondaryIndex.IndexName));
 
-                if (!string.IsNullOrEmpty(ttlAttributeName))
+                foreach (var secondaryIndex in secondaryIndexesToCreate)
                 {
-                    await ddbClient.UpdateTimeToLiveAsync(new UpdateTimeToLiveRequest
-                    {
-                        TableName = tableName,
-                        TimeToLiveSpecification = new TimeToLiveSpecification { AttributeName = ttlAttributeName, Enabled = true }
-                    });
+                    await TableCreateSecondaryIndex(tableDescription.TableName, attributes, secondaryIndex);
                 }
-                if (description.TableStatus != TableStatus.ACTIVE)
-                    throw new InvalidOperationException($"Failure creating table {tableName}");
             }
             catch (Exception exc)
             {
-                Logger.Error(ErrorCode.StorageProviderBase, $"Could not create table {tableName}", exc);
+                Logger.Error(ErrorCode.StorageProviderBase, $"Could not update table {tableDescription.TableName}", exc);
                 throw;
             }
         }
 
+        private async Task TableCreateSecondaryIndex(string tableName, List<AttributeDefinition> attributes, GlobalSecondaryIndex secondaryIndex)
+        {
+            await ddbClient.UpdateTableAsync(new UpdateTableRequest
+            {
+                TableName = tableName,
+                GlobalSecondaryIndexUpdates = new List<GlobalSecondaryIndexUpdate>
+                {
+                    new GlobalSecondaryIndexUpdate
+                    {
+                        Create = new CreateGlobalSecondaryIndexAction()
+                        {
+                            IndexName = secondaryIndex.IndexName,
+                            Projection = secondaryIndex.Projection,
+                            ProvisionedThroughput = secondaryIndex.ProvisionedThroughput,
```

Comment:
ProvisionedThroughput should be taken from the parent table configuration.

Reply:
Good spot!